src/main/java/org/ow2/proactive/catalog/rest/controller/BucketController.java

```diff
@@ -212,7 +212,14 @@ public BucketMetadata getMetadata(
                 String bucketGrantAccessType = bucketGrantService.getHighestGrantAccessTypeFromBucketGrants(user,
                                                                                                             data.getName(),
                                                                                                             data.getOwner());
-                BucketMetadata metadata = new BucketMetadata(data.getName(), data.getOwner(), data.getObjectCount());
+                int objectCount;
```

Comment:
This addition is to show exactly how many object are accessible for the user inside a bucket in case of object grants only or show the number of objects in case of bucket grant only (or bucket and object grant combined since this combination will allow the user to access all objects).
Before the number of all objects was showing all the tile regarding the case of only object grants.

Reply:
it's kind of weird that if grantCount is 0 you return the total number of objects.

For example, a user could have access to a bucket containing one object, but no access to this object. In which case, it should display 0 and not the total object count

Reply:
When you reach this verification, it's mean you already have a grant access to the bucket or a default access if you are the owner. I can  improve it to return directly the correct number. However the no access will be added when the negative grants are implemented.

Reply:
I understand, but to avoid forgetting it when you introduce the noaccess right, I would suggest to return the correct value.

At any rate, you save an extra call to getObjectCount(), which seems unnecessary. Unless this call is required in some cases ?

Reply:
Ok I'm fixing it. Usually grant count is 0 when there is no objects grants assigned to the bucket or if there is a bucket grant for the bucket. So when grant count is not 0, it means the user has access to the current bucket from object grants only, therefore, I return only the number of accessible objects. If if it's not the cas I return the number of all objects because the bucket is either public or accessible via bucket grant.

I'll try to fix it with out using if else. Directly return the correct number.

Reply:
thanks